src/v/transform/rpc/client.cc

```diff
@@ -96,6 +97,46 @@ cluster::errc map_errc(std::error_code ec) {
     }
     return cluster::errc::timeout;
 }
+
+template<typename Func>
+std::invoke_result_t<Func> retry(Func func) {
```

Comment:
Yeah sorry I added this in pieces.

Reply:
AFAIK we can't really forward the function because we call it multiple times, so if it was an rvalue then can't move it into futurize_invoke because the next iteration of the loop would be a use-after-move?